src/metrics/moves.ts

```diff
@@ -106,3 +106,106 @@ export class PieceLevelMoveInfoMetric implements Metric {
+  knightHops: UAPMap<{ count: number }>; // the number of times a piece is hopped over by knights
+
+  constructor() {
+    this.clear();
+  }
+
+  clear(): void {
+    this.enPassantMoves = 0;
+    this.knightHops = createUAPMap({ count: 0 });
+  }
+
+  processGame(
+    game: { move: PrettyMove; board: Piece[] }[],
+    metadata?: string[]
+  ) {
+    // since a knight can take 2 paths to hop over a piece we ensure that it is forced to hop over at least one piece.
+    // When it must hop we assume it hops all pieces so that the result is deterministic
+    for (const { move, board } of game) {
+      if (move.flags === 'e') {
+        this.enPassantMoves++;
+      }
+
+      if (move.piece === 'n') {
+        // a knight moving in a given direction has 4 things it can do in context of board index. It can increase by 1, decrease by 1, increase by 16, or decrease by 16.
+        // Right and down both increase total value, left and up decrease total value
+        // But it must do one of those things 2x. Meaning possible differences are 31, 33, 14, 18, -31, -33, -14, -18 (absolute values)
+        const moveDiff = move.toIndex - move.fromIndex;
+
+        // The offsets from current index to check for pieces hopped
+        let path1 = [];
+        let path2 = [];
+        switch (moveDiff) {
+          case 33:
+            path1 = [1, 17];
+            path2 = [16, 32];
+            break;
+          case 31:
+            path1 = [-1, 15];
+            path2 = [16, 32];
+            break;
+          case 18:
+            path1 = [1, 2];
+            path2 = [16, 17];
+            break;
+          case 14:
+            path1 = [-1, -2];
+            path2 = [16, 15];
+            break;
+          case -14:
+            path1 = [1, 2];
+            path2 = [-16, -15];
+            break;
+          case -18:
+            path1 = [-1, -2];
+            path2 = [-16, -17];
+            break;
+          case -31:
+            path1 = [1, -15];
+            path2 = [-16, -32];
+            break;
+          case -33:
+            path1 = [-1, -17];
+            path2 = [-16, -32];
+            break;
+          default:
+            throw new Error('unexpected moveDiff');
+        }
+        path1 = path1.map((offset) => move.fromIndex + offset);
+        path2 = path2.map((offset) => move.fromIndex + offset);
+        if (
+          (!board[path1[0]] && !board[path1[1]]) ||
+          (!board[path2[0]] && !board[path2[1]])
+        ) {
+          // if either path is clear we don't increment. Otherwise we randomly select a path to increment
+          continue;
+        } else {
+          // Could choose one path or the other at random. This makes a non-deterministic results
+          // const path = Math.random() > 0.5 ? path1[0] : path2[0];
+          for (const squareIndex of path1.concat(path2)) {
+            if (board[squareIndex]) {
+              this.knightHops[board[squareIndex].uas].count++;
+            }
```

Comment:
why are you concatenating path 1 squareindices to path2? wouldn't this currently potentially mean 4 pieces hopped per move, which we don't want?

i understand the discussion re:deterministic results, but i think it should still max out at 2 pieces hopped per move. this isn't schrodinger's cat lmao

if both paths contain pieces to hop, just pick the one with more pieces to hop imo

Reply:
Reasoning was: 
- Want unbiased result
- Want deterministic result

This metric will be viewed in the aggregate. When we aggregate we can divide by 2. This will remove any decision bias based on even/odd (i.e. common opening patterns frequently hopping) and produce a deterministic result for testing.

Reply:
choosing the one with more would also bias the results

Reply:
sg, i like it, please add to readme this explanation